lib/rubocop/cop/rails/where_missing.rb

```diff
@@ -0,0 +1,108 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # Check for the use `left_joins` and `where` method for finding missing relationship records.
+      #
+      # This cop is enabled in Rails 6.1 or higher.
+      #
+      # @example
+      #   # bad
+      #   Post.left_joins(:author).where(authors: { id: nil })
+      #
+      #   # good
+      #   Post.where.missing(:author)
+      #
+      class WhereMissing < Base
+        include RangeHelp
+        extend AutoCorrector
+        extend TargetRailsVersion
+
+        MSG = 'Use `where.missing(:%<association>s)` instead of ' \
+              '`%<left_joins_method>s(:%<association>s).where(%<association>ss: { id: nil })`.'
+        RESTRICT_ON_SEND = %i[left_joins left_outer_joins].freeze
+
+        minimum_target_rails_version 6.1
+
+        # @!method where_node_and_argument(node)
+        def_node_search :where_node_and_argument, <<~PATTERN
+          $(send ... :where (hash <(pair $(sym _) (hash (pair (sym :id) (nil))))...> ))
+        PATTERN
+
+        # @!method missing_relationship?(node)
+        def_node_search :missing_relationship?, <<~PATTERN
+          (pair (sym _) (hash (pair (sym :id) (nil))))
+        PATTERN
+
+        def on_send(node)
+          return unless node.first_argument.sym_type?
+
+          where_node_and_argument(node.ancestors.last) do |where_node, where_argument|
+            next unless same_relationship?(where_argument, node.first_argument)
+
+            range = range_between(node.loc.selector.begin_pos, node.loc.expression.end_pos)
+            register_offense(node, where_node, range)
+            break
+          end
+        end
+
+        private
+
+        def same_relationship?(where, left_joins)
+          "#{left_joins.value}s" == where.value.to_s
+        end
+
+        def register_offense(node, where_node, range)
+          add_offense(range, message: message(node)) do |corrector|
+            corrector.replace(node.loc.selector, 'where.missing')
+            if multi_condition?(where_node.first_argument)
+              replace_where_method(corrector, where_node)
+            else
+              remove_where_method(corrector, node, where_node)
+            end
+          end
+        end
+
+        def replace_where_method(corrector, where_node)
+          where_node.first_argument.children.each do |child|
+            next unless missing_relationship?(child)
+
+            corrector.remove(replace_range(child))
+          end
```

Comment:
it's a node_search so we can just apply it to the where_node
```suggestion
          missing_relationship(where_node) do |where_clause|
            corrector.remove(replace_range(where_clause))
          end
```

Reply:
I see. This looks good. I updated it.